orttraining/orttraining/test/python/orttraining_test_pipeline_parallel_tests.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import sys
+import argparse
+
+from _test_commons import run_subprocess
+
+import logging
+
+logging.basicConfig(
+    format="%(asctime)s %(name)s [%(levelname)s] - %(message)s",
+    level=logging.DEBUG)
+log = logging.getLogger("Build")
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--cwd", help="cwd")
+    return parser.parse_args()
+
+
+def main():
+    import torch
+    ngpus = torch.cuda.device_count()
+
+    # TODO: currently the CI machine only has 4 GPUs for parallel tests.
+    # Fill in more pipeline partition options when the machine has different GPUs counts.
+    if ngpus != 4:
+        return 0
```

Comment:
Is it required to have 4 gpus for the test?

Reply:
The current test requires 4 GPUs, not more, not less. I rewrote the test code a bit to reflect this. If we have less then 4 GPUs, we skip the test. Otherwise, the test is run.

---
In reply to: [549430310](https://github.com/microsoft/onnxruntime/pull/5815#discussion_r549430310) [](ancestors = 549430310)